hibernate-core/src/main/java/org/hibernate/tool/schema/extract/internal/InformationExtractorJdbcDatabaseMetaDataImpl.java

```diff
@@ -282,7 +282,7 @@ private TableInformation locateTableInNamespace(
 
 		if ( extractionContext.getJdbcEnvironment().getNameQualifierSupport().supportsCatalogs() ) {
 			if ( catalog == null ) {
-				catalogFilter = "";
+				catalogFilter = null;
```

Comment:
My only concern is not to break the behavior for some other databases. I think we should have a test case to replicate this issue on SQL Server. There are several examples in hibernate-core that use @RequireDialect + SQLServerDialect. This way I can run the test against other DBs and see how it behaves.